htop.c

```diff
@@ -53,6 +55,25 @@ static void printHelpFlag() {
    exit(0);
 }
 
+static bool parsePID(char *pid_str, unsigned int *result) {
+   char *endptr;
+   unsigned long tmp_result;
+
+   if (pid_str[0] == '-')
+      return false;
+
+   errno = 0;
+   tmp_result = strtoul(pid_str, &endptr, 10);
+   if (*endptr != '\0' || errno != 0)
+      return false;
+
+   if (tmp_result > UINT_MAX)
+      return false;
```

Comment:
this test is also unnecessary, because this doesn't really check for the range of valid PIDs anyway. Given any valid number can be a non-existent PID, I don't think it's necessary or useful to validate PID by its numeric value.

As we suggested before, it's better to simplify all these tests, not just move them all over to a separate function.